src/diggler/command.d

```diff
@@ -422,22 +422,66 @@ abstract class CommandSet(T) : ICommandSet
 
 	final void registerCommands(T cmdSet)
 	{
-		foreach(memberName; __traits(derivedMembers, T))
+		import std.meta : Filter;
+		import std.algorithm;
+		import std.string;
+		import std.array;
+
+		enum isCommand(string symbol) =
+			symbol != "__ctor" &&
+			symbol != "__dtor" &&
+			__traits(compiles, __traits(getMember, T, symbol)) && // ahem...
+			__traits(getProtection, __traits(getMember, T, symbol)) == "public" &&
+			isSomeFunction!(mixin("T." ~ symbol)) &&
+			!__traits(isStaticFunction, mixin("T." ~ symbol)) &&
+			!hasAttribute!(mixin("T." ~ symbol), ignore);
+
+		enum isSub(string symbol) = symbol[0] != '_' && symbol.canFind("_");
+
+		string[] handled_cmds;
+		foreach(memberName ; Filter!(isSub, Filter!(isCommand, __traits(derivedMembers, T))))
 		{
-			static if(
-				memberName != "__ctor" &&
-				memberName != "__dtor" &&
-				__traits(compiles, __traits(getMember, T, memberName)) && // ahem...
-				__traits(getProtection, __traits(getMember, T, memberName)) == "public" &&
-				isSomeFunction!(mixin("T." ~ memberName)) &&
-				!__traits(isStaticFunction, mixin("T." ~ memberName)) &&
-				!hasAttribute!(mixin("T." ~ memberName), ignore))
+			enum parts = memberName.split("_");
+			if(!getCommand(parts[0]))
 			{
-				auto dg = &mixin("cmdSet." ~ memberName);
-				auto cmd = Command.create!(mixin("T." ~ memberName))(dg);
+				Command[] subCmds;
```

Comment:
No reason for this to dynamically grow, its final state is known at compile-time